### PR TITLE
Remove client bundles for AMP only pages

### PR DESCRIPTION
**Pull request by ijjk**
Since we don't use the client bundles for AMP only pages we remove them.

### CODE REVIEW

File: packages/next/build/index.ts [hunk 1]
```diff
@@ -206,14 +206,16 @@ export default async function build(dir: string, conf = null): Promise<void> {
...
+    const serverPage = path.join(distDir, config.target === 'serverless' ? 'serverless/pages' : `server/static/${buildId}/pages`, page + '.js')
+    const clientPage = path.join(distDir, 'static', buildId, 'pages', page + '.js')
+
+    try {
+      let mod = require(serverPage)
+      mod = mod.default || mod
+      if (mod && mod.__nextAmpOnly) {
+        await unlink(clientPage)
+      }
+    } catch (_) {}
```
Comment (timneutkens) [inline packages/next/build/index.ts]: This should check for the same errors as recursive delete

Comment (ijjk) [reply packages/next/build/index.ts]: Updated

**Review by timneutkens — APPROVED** [review]

File: packages/next/server/on-demand-entry-handler.js [hunk 2]
```diff
@@ -261,6 +263,21 @@ export default function onDemandEntryHandler (devMiddleware, multiCompiler, {
...
+          try {
+            require('next/config').setConfig({})
+            let mod = require(serverPage)
+            mod = mod.default || mod
+            if (mod && mod.__nextAmpOnly) {
+              fs.unlinkSync(clientPage)
+            }
+          } catch (err) {
+            if (err.code !== 'ENOENT' && (err.message && !err.message.includes('Cannot find module'))) {
+              reject(err)
```
Comment (timneutkens) [inline packages/next/server/on-demand-entry-handler.js]: `return reject(err)`?

Comment (ijjk) [reply packages/next/server/on-demand-entry-handler.js]: Updated

**Pull request merged**